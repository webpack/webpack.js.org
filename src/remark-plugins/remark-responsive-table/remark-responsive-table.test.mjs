import remark from 'remark';
describe('responsive table', () => {
  const processor = remark()
    .use(require('remark-gfm'))
    .use(require('./remark-responsive-table.js'))
    .use(require('remark-html'));
  it('should add data-th', () => {
    processor.process(
      `
| foo | bar |
| --- | --- |
| baz |  |
| sam | chen |
      `,
      (error, { contents }) => {
        expect(error).toBeNull();
        expect(contents).toContain('data-th="foo"');
        expect(contents).toContain('data-th="bar"');
        expect(contents).toContain('-');
        expect(contents).toMatchSnapshot();
      }
    );
  });
  it('should handle empty thead', () => {
    processor.process(
      `
| | bar |
| --- | --- |
| baz |  |
| sam | chen |
      `,
      (error, { contents }) => {
        expect(error).toBeNull();
        expect(contents).toMatchSnapshot();
      }
    );
  });
});
