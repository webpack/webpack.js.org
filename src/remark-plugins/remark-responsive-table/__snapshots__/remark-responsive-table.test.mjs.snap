// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`responsive table should add data-th 1`] = `
"<table>
<thead>
<tr>
<th>foo</th>
<th>bar</th>
</tr>
</thead>
<tbody>
<tr>
<td data-th=\\"foo\\"><span>baz</span></td>
<td data-th=\\"bar\\"><span>-</span></td>
</tr>
<tr>
<td data-th=\\"foo\\"><span>sam</span></td>
<td data-th=\\"bar\\"><span>chen</span></td>
</tr>
</tbody>
</table>
"
`;

exports[`responsive table should handle empty thead 1`] = `
"<table>
<thead>
<tr>
<th></th>
<th>bar</th>
</tr>
</thead>
<tbody>
<tr>
<td data-th=\\"\\"><span>baz</span></td>
<td data-th=\\"bar\\"><span>-</span></td>
</tr>
<tr>
<td data-th=\\"\\"><span>sam</span></td>
<td data-th=\\"bar\\"><span>chen</span></td>
</tr>
</tbody>
</table>
"
`;
